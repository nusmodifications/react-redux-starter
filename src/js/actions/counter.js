// @flow
import type { FSA } from 'types/redux';

export const INCREMENT = 'INCREMENT';
export function increment(): FSA {
  return {
    type: INCREMENT,
    payload: null,
  };
}

export const DECREMENT = 'DECREMENT';
export function decrement(): FSA {
  return {
    type: DECREMENT,
    payload: null,
  };
}
