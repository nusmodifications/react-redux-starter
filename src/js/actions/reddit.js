// @flow

import { API_REQUEST } from 'middlewares/requests-middleware';

export const FETCH_REDDITS: string = 'FETCH_REDDITS';
export function fetchReddits(topic: string): Function {
  return (dispatch: Function) => dispatch({
    [API_REQUEST]: {
      type: FETCH_REDDITS,
      payload: {
        method: 'GET',
        url: `https://www.reddit.com/r/${topic}.json`,
      },
    },
  });
}
