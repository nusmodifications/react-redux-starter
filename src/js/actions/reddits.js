// @flow
import type { FSA } from 'types/redux';
import { API_REQUEST } from 'middlewares/requests-middleware';

export const FETCH_REDDITS: string = 'FETCH_REDDITS';
export function fetchReddits(topic: string): FSA {
  return {
    type: FETCH_REDDITS,
    payload: {
      method: 'GET',
      url: `https://www.reddit.com/r/${topic}.json`,
    },
    meta: {
      [API_REQUEST]: true,
    },
  };
}
