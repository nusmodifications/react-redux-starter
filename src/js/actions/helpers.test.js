import type { FSA } from 'types/redux';
import * as actions from 'actions/helpers';

describe('helpers actions', () => {
  describe('resetRequestState', () => {
    it('should return an action', () => {
      const payload: Array<any> = [];
      const action: FSA = actions.resetRequestState(payload);
      expect(action).toMatchSnapshot();
    });

    it('should convert an element into an array', () => {
      const payload: string = 'TEST_ACTION';
      const action: FSA = actions.resetRequestState(payload);
      expect(action).toMatchSnapshot();
    });
  });

  describe('resetErrorState', () => {
    it('should return an action', () => {
      const payload: Array<any> = [];
      const action: FSA = actions.resetErrorState(payload);
      expect(action).toMatchSnapshot();
    });

    it('should convert an element into an array', () => {
      const payload: string = 'TEST_ACTION';
      const action: FSA = actions.resetErrorState(payload);
      expect(action).toMatchSnapshot();
    });
  });

  describe('resetAllState', () => {
    it('should return an action', () => {
      const action: FSA = actions.resetAllState();
      expect(action).toMatchSnapshot();
    });
  });
});
