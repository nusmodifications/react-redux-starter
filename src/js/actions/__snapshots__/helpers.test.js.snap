// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers actions resetAllState should return an action 1`] = `
Object {
  "payload": null,
  "type": "RESET_ALL_STATE",
}
`;

exports[`helpers actions resetErrorState should convert an element into an array 1`] = `
Object {
  "payload": Array [
    "TEST_ACTION",
  ],
  "type": "RESET_ERROR_STATE",
}
`;

exports[`helpers actions resetErrorState should return an action 1`] = `
Object {
  "payload": Array [],
  "type": "RESET_ERROR_STATE",
}
`;

exports[`helpers actions resetRequestState should convert an element into an array 1`] = `
Object {
  "payload": Array [
    "TEST_ACTION",
  ],
  "type": "RESET_REQUEST_STATE",
}
`;

exports[`helpers actions resetRequestState should return an action 1`] = `
Object {
  "payload": Array [],
  "type": "RESET_REQUEST_STATE",
}
`;
